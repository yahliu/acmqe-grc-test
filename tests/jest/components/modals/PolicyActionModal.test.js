/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import React from 'react'
import { PolicyActionModal } from '../../../../src-web/components/modals/PolicyActionModal'
import * as reducers from '../../../../src-web/reducers'
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { AcmAlert } from '@open-cluster-management/ui-components'
import { Spinner } from '@patternfly/react-core'

import {
  disableResourceModalLabels, enableResourceModalLabels,
  enforceResourceModalLabels, informResourceModalLabels  } from './ModalsTestingData'
import { REQUEST_STATUS } from '../../../../src-web/actions/index'

describe('PolicyActionModal component', () => {
  it('Disable modal renders as expected', () => {
    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        type='resource-disable'
        handleClose={jest.fn()}
        handleSubmit={jest.fn()}
        label={disableResourceModalLabels}
        locale={'en'}
        open={true}
        store={store}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Enable modal renders as expected', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        type='resource-enable'
        handleClose={jest.fn()}
        handleSubmit={jest.fn()}
        label={enableResourceModalLabels}
        locale={'en'}
        open={true}
        store={store}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Enforce modal renders as expected', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        type='resource-enforce'
        handleClose={jest.fn()}
        handleSubmit={jest.fn()}
        label={enforceResourceModalLabels}
        locale={'en'}
        open={true}
        store={store}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Inform modal renders as expected', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        type='resource-inform'
        handleClose={jest.fn()}
        handleSubmit={jest.fn()}
        label={informResourceModalLabels}
        locale={'en'}
        open={true}
        store={store}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Modal renders without data', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        label={{}}
        locale={'en'}
        open={true}
        store={store}
      />
    )
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Modal returns spinner when loading', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        label={{}}
        locale={'en'}
        open={true}
        reqStatus={REQUEST_STATUS.IN_PROGRESS}
        store={store}
      />
    )
    expect(component.find(Spinner)).toHaveLength(1)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('Modal returns notification on error', () => {

    const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
    const middleware = [thunkMiddleware]
    const store = createStore(combineReducers(reducers), composeEnhancers(
      applyMiddleware(...middleware)
    ))
    const component = shallow(
      <PolicyActionModal
        label={{}}
        locale={'en'}
        open={true}
        reqStatus={REQUEST_STATUS.ERROR}
        reqErrorMsg='There was an error.'
        store={store}
      />
    )
    expect(component.find(AcmAlert)).toHaveLength(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
