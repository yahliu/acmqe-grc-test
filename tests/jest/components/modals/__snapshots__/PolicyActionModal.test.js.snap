// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicyActionModal component Disable modal renders as expected 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        >
          Disable policy
        </AcmButton>,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id="disable-resource-modal"
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    title="Disable policy"
    titleIconVariant="warning"
    variant="medium"
  >
    <div />
    Are you sure you want to disable this policy?
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Enable modal renders as expected 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        >
          Enable policy
        </AcmButton>,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id="enable-resource-modal"
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    title="Enable policy"
    titleIconVariant="default"
    variant="medium"
  >
    <div />
    Are you sure you want to enable this policy?
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Enforce modal renders as expected 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        >
          Enforce policy
        </AcmButton>,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id="enforce-resource-modal"
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    title="Enforce policy"
    titleIconVariant="warning"
    variant="medium"
  >
    <div />
    Are you sure you want to enforce this policy? The policy makes any necessary changes to the applied cluster to remediate the violations.
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Inform modal renders as expected 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        >
          Inform policy
        </AcmButton>,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id="inform-resource-modal"
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    title="Inform policy"
    titleIconVariant="default"
    variant="medium"
  >
    <div />
    Are you sure you want to change the remediation action of this policy to inform? You must manually remediate the policy violations.
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Modal renders without data 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        />,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id=""
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    titleIconVariant="default"
    variant="medium"
  >
    <div />
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Modal returns notification on error 1`] = `
<div>
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        />,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id=""
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    titleIconVariant="default"
    variant="medium"
  >
    <div>
      <AcmAlert
        isInline={true}
        noClose={true}
        title="There was an error."
        variant="warning"
      />
    </div>
  </AcmModal>
</div>
`;

exports[`PolicyActionModal component Modal returns spinner when loading 1`] = `
<div>
  <Spinner
    className="patternfly-spinner"
  />
  <AcmModal
    actions={
      Array [
        <AcmButton
          onClick={[Function]}
          variant="primary"
        />,
        <AcmButton
          onClick={[Function]}
          variant="link"
        >
          Cancel
        </AcmButton>,
      ]
    }
    id=""
    isOpen={true}
    onClose={[Function]}
    showClose={true}
    titleIconVariant="default"
    variant="medium"
  >
    <div />
  </AcmModal>
</div>
`;
